### ISSUE TITLE
Improve ManhattanDistance metric for DDP 

### ISSUE THREAD

**Issue by sdesrozis** [pytorch/ignite#1319]
## 🚀 Feature

Current formula of `ManhattanDistance` is not correct, it misses an absolute value. The idea is to use `sklearn` to ensure good computation. 

Current implementation of `ManhattanDistance` metric is not compatible with ddp. The idea is to improve this metric to work in parallel.

Related to #1284
